.github/workflows/merge.yml

```diff
@@ -0,0 +1,55 @@
+name: Merge
+
+on:
+  push:
+    branches: [ "main" ]
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
```

Comment:
I added a few comments here, but they are generally also applicable to the main pull request GitHub action. 

It might even make sense to ensure that we don't duplicate this by just calling the PR GitHub action in its completeness and move almost everything except for the documentation deployment over there? 

E.g., we do a similar thing in our Spezi iOS Repos: https://github.com/StanfordSpezi/SpeziTemplateApplication/blob/fafb5c6c30f9e049b6515e7ea9f46af624c309a1/.github/workflows/beta-deployment.yml#L18

We could simply do something like
```yml
buildandtest:
    name: Build and Test
    uses: ./.github/workflows/pull_request.yml
    permissions:
      security-events: write
      packages: read
      actions: read
      contents: read
    secrets: inherit
```

Reply:
liked this a lot 🚀 TIL moment 💯 

Applied and will observe after merging